src/bot/root_pythia_cogs.py

```diff
@@ -63,7 +63,30 @@ async def resume(self, ctx):
             "Resumed successfully from idle state, requests can be sent again."
         )
 
-    # TODO: add a add_users command that would accept a list of ids
+    @commands.before_invoke(log_command_call)
+    @commands.command(name="addusers")
+    async def addusers(self, ctx, *args):
+        self.logger.info(f"command `addusers` received: {len(args)} arguments:")
+        for i in range(len(args)):
+            try:
+                user_id = int(args[i])
+                user = await self.dbmanager.add_user(user_id)
+            except ValueError as value_err: # Error coming from the int() cast
+                self.logger.error(f"command `addusers` received: {value_err}")
+                await ctx.message.channel.send(f"invalid argument received: an user_id is expected, ignoring it for now...")
+                continue
+            # TODO: except 404 error -> if the request in add_user fails, we should continue here (ex: multiple users but only one can be wrong)
+            # except 404Error as 404_err: # Error coming from add_user() method
+            #     pass
+            
+            if user is None:
+                await ctx.message.channel.send(f"UserID {user_id} already exists in database")
+                self.logger.warning(f"UserID {user_id} already exists in database")
+                continue
+
+            await ctx.message.channel.send(f"{user} added!\nPoints: {user.score}")
+            self.logger.info("command `addusers` added user '%s'", user)
```

Comment:
maybe we could factorize this code with the code from `adduser`? To enforce consistency in logging and messaging between both commands
or just use predefined strings would work too, I don't know

but as you want :shrug:

Reply:
Not sure to understand what you mean by `consistency in logging and messaging`. I feel like a function for this block of code is too specific. All the more, giving strings as parameter for logging looks pretty ugly to me.

Reply:
> Not sure to understand what you mean by consistency in logging and messaging

Only that there is some code duplication between `adduser` and `addusers`
Maybe a function (not a command) like `add_one_user` which would be called once by `adduser` and multiple times by `addusers` would do the job.
But maybe you feel that this is too complex for not enough benefit and in thta case feel free to tell me!

> All the more, giving strings as parameter for logging looks pretty ugly to me.

you're right about that, I was speaking of defining (as static class attr for example) a `ERR_MSG_USER_IN_DB="UserID %s already exists in database"` to be used in each command as `await ctx.message.channel.send(ERR_MSG_USER_IN_DB % user_id)` which would enforce to have the same msg in each command! But that's just an idea

Reply:
>Maybe a function (not a command) like add_one_user which would be called once by adduser and multiple times by addusers would do the job.

I totally agree with that idea. I just didn't know if it was possible to call an `async` function in another. Same as the function `adduser` has decorators. Do you happen to know if it is possible to do that ?

For the second point, you are right.  If you tell me it is possible to call `adduser` in `addusers`, I will factorize the code :wink:

Reply:
> I totally agree with that idea. I just didn't know if it was possible to call an async function in another.

of course you can:
```py
async def child(b):
  # some code


async def parent(a):
  # some code
  await child(...)
```
is perfectly valid code, only one thing: an `async` function **must be `await`ed**
That's what we're doing when we send msg with `await ctx....`: we're calling another async method!

------------------

> Same as the function adduser has decorators. Do you happen to know if it is possible to do that ?

>  If you tell me it is possible to call adduser in addusers, I will factorize the code wink

Decorators are strictly speaking just function that returns another function, thus overwriting/wrapping the original function (`adduser` for example) see it the way you prefer (check this excellent tuto if you'd like https://www.geeksforgeeks.org/decorators-in-python/).
For example the `check_new_solves` function decorated by `tasks.loop` is an instance of the Task class, and I then call its method like `check_new_solves.restart()` in `loop_error_handler`, it **has been replaced**.

So it depends on the decorator :upside_down_face: 
In our case we'd have to dig in the `discord.py` [doc](https://discordpy.readthedocs.io/en/stable/interactions/api.html?highlight=commands%20command#discord.app_commands.command) to know if/by what the function is replaced.
But intuitively I'd say that we can't :confused:. That's why I suggested to factorize this code in another different function `add_one_user`:
> Maybe a function (not a command) like add_one_user which would be called once by adduser and multiple times by addusers would do the job

Reply:
Thanks for your very detailled answer ! Sorry for the delay. It wasn't long too fix but I didn't have much time lately. 

Anyway, I factorize `adduser` code into one `async function` called `base_add_one_user` which is called in both `adduser` and `addusers` command. Hope you will like it !

The behaviour of the bot has not changed since my last tests.